### PR TITLE
feat: format requested files, not everything

### LINKED ISSUE #4: Accept list of files to format as positional command-line args
Some tools like pre-commit expect formatter tools to be called like `formatter file.fmt`.

This is nice because that tool has already determined which files are modified by the user in a given context, so we won't have to wait to "re-format the world".

### PR DESCRIPTION
**Pull request by alexeagle**
fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,20 +60,36 @@ copy the WORKSPACE snippet into your `WORKSPACE` file.
...
-Check that files are already formatted, exit non-zero if formatting is needed:
+⚡ Install under pre-commit.com hook, in your `.pre-commit-config.yaml`:
+
+```yaml
+- repo: local
+  hooks:
+    - id: bazel-super-formatter
+      name: Format
+      language: system
+      entry: bazel run -- //:format fix
```
Comment (cgrindel) [inline README.md:86]: Just to confirm. I always thought that the syntax was `bazel run //:format -- fix` where the double dash follows the target.

**Review by cgrindel — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -60,20 +60,36 @@ copy the WORKSPACE snippet into your `WORKSPACE` file.
 
 ## Usage
 
-One-time re-format all files:
+⚡ One-time re-format all files:
```
Comment (jbedard) [inline README.md:63]: ⚡ fast?

File: format/format.sh [hunk 3]
```diff
@@ -63,28 +67,44 @@ case "$mode" in
  *) echo >&2 "unknown mode $mode";;
 esac
 
-files=$(git ls-files '*/BUILD.bazel' '*.bzl' '*.BUILD' 'WORKSPACE' '*.bazel')
+if [ "$#" -eq 0 ]; then
+  files=$(git ls-files '*/BUILD.bazel' '*.bzl' '*.BUILD' 'WORKSPACE' '*.bazel')
+else
+  files=$(find "$@" -name '*/BUILD.bazel' -or -name '*.bzl' -or -name '*.BUILD' -or -name 'WORKSPACE' -or -name '*.bazel')
```
Comment (jbedard) [inline format/format.sh:73]: Does this need to respect `.gitignore`?

Comment (jbedard) [reply format/format.sh:73]: Oh, this just filters the files from the cli args?

**Review by jbedard — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (alexeagle) [reply README.md:86]: you're right - if you have no "negative target patterns" like `-//other/...` then it doesn't matter, but your way is more idiomatic for it to follow the targets

File: README.md (hunk 2 above)
Comment (alexeagle) [reply README.md:63]: I dunno, it was just boring-looking so I needed an emoji. 💎 maybe?

File: format/format.sh (hunk 3 above)
Comment (alexeagle) [reply format/format.sh:73]: correct

**Pull request merged**